v-next/hardhat/test/internal/builtin-plugins/network-manager/type-validation.ts

```diff
@@ -0,0 +1,396 @@
+import type {
+  EdrNetworkConfig,
+  HttpNetworkConfig,
+} from "@ignored/hardhat-vnext/types/config";
+
+import assert from "node:assert/strict";
+import { describe, it } from "node:test";
+
+import { validateNetworkConfig } from "../../../../src/internal/builtin-plugins/network-manager/type-validation.js";
+
+describe("HttpNetworkConfig", () => {
+  describe("accounts", function () {
+    const httpNetworkConfig: HttpNetworkConfig = {
+      type: "http",
+      gas: "auto",
+      gasMultiplier: 1,
+      gasPrice: "auto",
+      accounts: "remote", // Modified in the tests
+      url: "http://localhost:8545",
+      timeout: 20_000,
+      httpHeaders: {},
+    };
+
+    describe("allowed values", () => {
+      it("should allow the value 'remote'", function () {
+        httpNetworkConfig.accounts = "remote";
+
+        const validationErrors = validateNetworkConfig(httpNetworkConfig);
+
+        assert.equal(validationErrors.length, 0);
+      });
+
+      it("should allow an array of valid private keys", function () {
+        httpNetworkConfig.accounts = [
+          "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+          "0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb",
+          "0xcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc",
+        ];
+
+        const validationErrors = validateNetworkConfig(httpNetworkConfig);
+
+        assert.equal(validationErrors.length, 0);
+      });
+
+      it("should allow an account with a valid HttpNetworkHDAccountsConfig", function () {
+        httpNetworkConfig.accounts = {
+          mnemonic: "asd asd asd",
+          initialIndex: 0,
+          count: 123,
+          path: "m/123",
+          passphrase: "passphrase",
+        };
+
+        const validationErrors = validateNetworkConfig(httpNetworkConfig);
+
+        assert.equal(validationErrors.length, 0);
+      });
+    });
+
+    describe("not allowed values", () => {
+      describe("wrong private key formats", () => {
+        // TODO: still valid?
```

Comment:
Do we still wanna keep the possibility of allowing valid private keys with missing hex prefix?
If yes, I will have to modify the hh-utils method that I added to check if an address is valid, [see here](https://github.com/NomicFoundation/hardhat/pull/5817/files#diff-8d9a1ba6601820ff6249438ad3797d59980861eef0ca459e2a05512362894bcd)